core/domain/cron_services.py

```diff
@@ -121,3 +122,39 @@ def reduce(key, stringified_values):
             logging.warning(
                 'Entities remaining count: %s entities (%s)' %
                 (sum(values), key))
+
+
+class JobModelsCleanupManager(jobs.BaseMapReduceOneOffJobManager):
+    """One-off job for cleaning up old entities of JobModel."""
+
+    TWELVE_WEEKS = datetime.timedelta(weeks=12)
+    STATUSES_TO_DELETE = [
+        job_models.STATUS_CODE_COMPLETED,
+        job_models.STATUS_CODE_FAILED,
+        job_models.STATUS_CODE_CANCELED
+    ]
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        """The entity types this job will handle."""
+        return [job_models.JobModel]
+
+    @staticmethod
+    def map(model):
+        date_twelve_weeks_ago = (
```

Comment:
Should we change the naming here such that it doesn't include 12? That way in future if we planned to change 12 to 10 then we just have to do changes in one place*

[I don't have suggestion for better naming! :p ]

Reply:
I thought about it (when I was writing the code) but it seemed to me it would be better to write this explicitly and if needed change the name of the variable too.

Reply:
I generally prefer having general constant name such that it's easier to change the contact value in the future without touching any other part of the code. Considering the scope of this constant is small I'm fine with it! Thanks!